sessionize/lib/src/commonMain/kotlin/co/touchlab/sessionize/architecture/MainThreadPubSub.kt

```diff
@@ -15,14 +18,24 @@ class MainThreadPubSub<T> : BasePub<T>(), Sub<T> {
     override fun subs(): MutableCollection<Sub<T>> = subSetLocal.lateValue
 
     override fun onNext(next: T) {
-        backToFront({ next.freeze() }) {
-            applyNextValue(it)
+        next.freeze()
+        if(ServiceRegistry.concurrent.allMainThread){
```

Comment:
Minor structure comment, I don't think it warrants a change before merging: Would it make sense to internalize this kind of checking in the Concurrent class so that it manages the whole background->foreground question without callers needing to worry about it?